shell/browser/ui/file_dialog_gtk.cc

```diff
@@ -101,14 +202,23 @@ class FileChooserDialog {
     if (!settings.filters.empty())
       AddFilters(settings.filters);
```

Comment:
`AddFilters` uses `gtk_file_filter_add_custom`, which is skipped when using the portal. This actually breaks things badly, as GTK will include a filter list when it sees a filter is present, but it ends up empty due to using all the custom filters, resulting in the portal rejecting the call altogether.

Reply:
So you're saying to check "if GtkFileChooserNative, then don't AddFilters"?

Reply:
Not quite. AddFilters itself needs to be adjusted to use the normal string-based GtkFileFilter functions to add the patterns, instead of using add_custom, when using a native dialog. I mentioned it here since, err, it's the only part of the code related to filters that I could put a comment on 😅 though I definitely should have clarified more.

Reply:
I'll try to review what you have said to the PR. May have to come back to ask more questions :).